src/hypothesis/core.py

```diff
@@ -215,7 +215,7 @@ def simplify_template_such_that(
         any_simplifiers = False
         for simplify in search_strategy.simplifiers(random, t):
             debug_report(u'Applying simplification pass %s' % (
-                simplify.__name__,
+                unicode_safe_repr(simplify.__name__),
```

Comment:
I don't think this is the right thing to do. This is now showing an escaped string where it would previously be showing a name.

Something along the lines of a type_name() function in hypothesis.internal.compat which tries to turn __name__ into a unicode object if it's not already one is probably the way to go (something like 'decode as utf-8, if that fails show an escaped version' is probably appropriate)